model/result/ScoreWriterService.php

```diff
@@ -0,0 +1,130 @@
+use oat\taoResultServer\models\classes\ResultServerService;
+use taoResultServer_models_classes_OutcomeVariable as ResultServerOutcomeVariable;
+use Throwable;
+
+class ScoreWriterService extends ConfigurableService
+{
+    /**
+     * Store the score result into a delivery execution
+     *
+     * @param array $result
+     *
+     * @return string
+     * @throws Throwable
+     */
+    public function store(array $result)
+    {
+        $this->validateResult($result);
+
+        $deliveryExecution = $this->getDeliveryExecution($result);
+
+        /** @var ResultServerService $resultServerService */
+        $resultServerService = $this->getResultServerService();
+
+        $resultStorageService = $resultServerService->getResultStorage($result['sourcedId']);
+
+        $resultStorageService->storeTestVariable(
+            $result['sourcedId'],
+            $deliveryExecution->getDelivery()->getUri(),
+            $this->getScoreVariable($result['score']),
+            $deliveryExecution->getIdentifier()
+        );
+
+        return $deliveryExecution->getIdentifier();
+    }
+
+    /**
+     * Look for a DeliveryExecution and return it or throw an Exception
+     *
+     * @param array $result
+     *
+     * @return DeliveryExecution
+     * @throws ResultException
+     */
+    private function getDeliveryExecution($result)
+    {
+        try {
+            /** @var ServiceProxy $resultService */
+            $resultService = $this->getServiceLocator()->get(ServiceProxy::SERVICE_ID);
+            return $resultService->getDeliveryExecution($result['sourcedId']);
+        } catch (Throwable $exception) {
+            throw new ResultException(
+                $exception->getMessage(),
+                MessageBuilder::STATUS_DELIVERY_EXECUTION_NOT_FOUND,
+                $exception,
+                MessageBuilder::build(MessageBuilder::STATUS_DELIVERY_EXECUTION_NOT_FOUND, $result)
+            );
+        }
+    }
+
+    /**
+     * @return ResultServerService
+     */
+    private function getResultServerService()
+    {
+        return $this->getServiceLocator()->get(ResultServerService::SERVICE_ID);
+    }
+
+    /**
+     * @param string $score
+     *
+     * @return ResultServerOutcomeVariable
+     * @throws common_exception_InvalidArgumentType
+     */
+    private function getScoreVariable($score)
+    {
+        $scoreVariable = $this->getResultServiceOutcomeVariable();
+        $scoreVariable->setIdentifier('SCORE');
+        $scoreVariable->setCardinality(ResultServerOutcomeVariable::CARDINALITY_SINGLE);
+        $scoreVariable->setBaseType('float');
+        $scoreVariable->setEpoch(microtime());
+        $scoreVariable->setValue($score);
+
+        return $scoreVariable;
+    }
+
+    /**
+     * @param array $result
+     *
+     * @throws InvalidScoreException
+     * @throws ResultException
+     */
+    private function validateResult(array $result)
+    {
+        $validator = new ValidateScoreResult();
```

Comment:
using direct instantiation of the class (opposite to retrieving it through `ServiceLocator`) makes it impossible to mock it in unit test